### PR TITLE
BaSP-TG-27: projects create test get by id, get all and create

### PR DESCRIPTION
**Pull request by julianpiaggio**
Create tests for the project entity (get)
Create tests for the project entity (getAll)
Create tests for the project entity (create)

### CODE REVIEW

File: src/tests/projects.test.js [hunk 1]
```diff
@@ -1,5 +1,77 @@
...
+          rate: 5.2,
+        }],
+      };
+      const response = await request(app).post('/projects').send(projectsMooked);
+      expect(response.status).toBe(201);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Project created successfully!');
+    });
+
```
Comment (kolodani) [inline src/tests/projects.test.js:60]: you can add a describe here to separate the test, from success to errors

File: src/tests/projects.test.js [hunk 2]
```diff
@@ -1,5 +1,77 @@
...
+  describe('GET BY ID /projects/:id', () => {
+    test('should return status code 200', async () => {
+      const response = await request(app).get('/projects/6354c3046634d3f5d058bae8').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe(`${response.body.message}`);
+    });
+
```
Comment (kolodani) [inline src/tests/projects.test.js:30]: you can add a describe here to separate the test, from success to errors

**Review by kolodani — COMMENTED** [review]
great  work

File: src/tests/projects.test.js (hunk 2 above)
Comment (lmalvasia) [reply src/tests/projects.test.js:30]: This is not necessary, we could have only one describe per `method`.

File: src/tests/projects.test.js [hunk 3]
```diff
@@ -1,5 +1,77 @@
...
+    test('should return status code 400', async () => {
+      const response = await request(app).get('/projects/6354c2ee2c85bbe8fb51072').send();
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe(`${response.body.message}`);
```
Comment (lmalvasia) [inline src/tests/projects.test.js:36]: This is equal to expect(true).toBe(true) 😂 
You have to write the message directly as `string`. For example:
expect(response.body.message).toBe('Invalid id');

File: src/tests/projects.test.js [hunk 4]
```diff
@@ -1,5 +1,77 @@
...
+  describe('GET BY ID /projects/:id', () => {
+    test('should return status code 200', async () => {
+      const response = await request(app).get('/projects/6354c3046634d3f5d058bae8').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe(`${response.body.message}`);
+    });
+
+    test('should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/projects.test.js:31]: You have to write a correct description. I don't know why we are getting an status 400. For example:
```suggestion
    test('should return an error when the id is invalid', async () => {
```

File: src/tests/projects.test.js [hunk 5]
```diff
@@ -1,5 +1,77 @@
...
+        }],
+      };
+      const response = await request(app).post('/projects').send(projectsMooked);
+      expect(response.status).toBe(201);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Project created successfully!');
+    });
+
+    test('should not create a project and return status 400', async () => {
```
Comment (lmalvasia) [inline src/tests/projects.test.js:61]: Change the description and add the purpose why this is going to return an error

**Review by kolodani — APPROVED** [review]
nice work

**Review by matescarabino — APPROVED** [review]
All fixed

**Pull request merged**